chains/evm/contracts/ccvs/CCTPVerifier.sol

```diff
@@ -0,0 +1,479 @@
+      revert InvalidMessageTransmitterOnProxy(address(messageTransmitter), messageTransmitterOnProxy);
+    }
+
+    // Set the immutable state variables.
+    i_tokenMessenger = tokenMessenger;
+    i_messageTransmitterProxy = messageTransmitterProxy;
+    i_localDomainIdentifier = messageTransmitter.localDomain();
+    i_usdcToken = usdcToken;
+
+    // Approve the token messenger to burn the USDC token on behalf of this contract.
+    // The USDC token pool will be responsible for forwarding USDC it receives from the router to this contract.
+    i_usdcToken.safeIncreaseAllowance(address(i_tokenMessenger), type(uint256).max);
+
+    emit StaticConfigSet(
+      address(i_tokenMessenger), address(i_messageTransmitterProxy), address(i_usdcToken), i_localDomainIdentifier
+    );
+
+    _setDynamicConfig(dynamicConfig);
+  }
+
+  /// @inheritdoc ICrossChainVerifierV1
+  function forwardToVerifier(
+    MessageV1Codec.MessageV1 calldata message,
+    bytes32 messageId,
+    address, // feeToken
+    uint256, // feeTokenAmount
+    bytes calldata // verifierArgs
+  ) external returns (bytes memory verifierReturnData) {
+    // For EVM, sender is expected to be 20 bytes.
+    address senderAddress = address(bytes20(message.sender));
+    _assertSenderIsAllowed(message.destChainSelector, senderAddress);
+
+    Domain storage domain = s_chainToDomain[message.destChainSelector];
+    if (!domain.enabled) revert UnknownDomain(message.destChainSelector);
+
+    // We expect exactly one token transfer per message.
+    if (message.tokenTransfer.length != 1) revert InvalidTokenTransferLength(message.tokenTransfer.length);
+
+    // The address of the token transferred must correspond to USDC.
+    address sourceTokenAddress = address(bytes20(message.tokenTransfer[0].sourceTokenAddress));
+    if (sourceTokenAddress != address(i_usdcToken)) revert InvalidToken(sourceTokenAddress);
+
+    if (message.tokenTransfer[0].tokenReceiver.length > 32) {
+      revert InvalidReceiver(message.tokenTransfer[0].tokenReceiver);
+    }
+
+    bytes32 decodedReceiver;
+    // For EVM chains, the mintRecipient is not used.
+    // Solana requires it, as the mintRecipient will be a PDA owned by the pool.
+    // The PDA will forward the tokens to their final destination after minting.
+    if (domain.mintRecipientOnDest != bytes32(0)) {
+      decodedReceiver = domain.mintRecipientOnDest;
+    } else {
+      bytes memory receiver = message.tokenTransfer[0].tokenReceiver;
+      uint256 length = receiver.length;
+      // solhint-disable-next-line no-inline-assembly
+      assembly {
+        // Load the data, skipping the length word.
+        let data := mload(add(receiver, 32))
+        // Left-pad the data with (32-length) bytes.
+        // Right-shifting the data by (32-length) bytes * 8 bits per byte achieves this.
+        decodedReceiver := shr(mul(sub(32, length), 8), data)
+      }
+    }
+
+    _depositForBurn(
```

Comment:
At this point it's probably fine to inline this function? Saves passing the many params

Reply:
https://github.com/smartcontractkit/chainlink-ccip/pull/1396/commits/7215895e67b0bd33709f845d03ea2dd8ca937524